src/app/card/card.component.ts

```diff
@@ -16,25 +18,36 @@ export class MzCardComponent implements AfterViewInit {
   @Input() hoverable: boolean;
   @Input() textClass: string;
 
-  @ViewChild('cardActionWrapper') cardActionWrapper: ElementRef;
+  hasCardAction = true;
+  hasCardTitle = true;
 
-  constructor() { }
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    private elementRef: ElementRef,
+  ) {}
 
   ngAfterViewInit() {
-    this.initCardAction();
+    this.hasCardTitle = this.hasTitleTagAndNotEmpty();
+    this.hasCardAction = this.hasActionTagAndNotEmpty();
+    this.changeDetectorRef.detectChanges();
   }
 
-  private initCardAction() {
-    const cardActionElement = $(this.cardActionWrapper.nativeElement).find('mz-card-action');
+  private hasActionTagAndNotEmpty(): boolean {
+    const cardActionElement = this.elementRef.nativeElement.querySelector('mz-card-action');
 
-    // hide action wrapper when mz-card-action element is not present
-    // or when mz-card-action does not contain any html (text or elements)
-    if (cardActionElement.length === 0 || cardActionElement[0].innerHTML.trim() === '') {
-      this.cardActionWrapper.nativeElement.style.display = 'none';
-    }
+    return this.isElementDisplayed(cardActionElement);
+  }
+
+  private hasTitleTagAndNotEmpty(): boolean {
```

Comment:
It does not work with getter. The title and action are never displayed and I see the getters function are called very often.